extension.js

```diff
@@ -80,7 +80,7 @@ export default class DashToDockExtension extends Extension.Extension {
         if (!this._isEnabled || this._statusNotifierWatcher)
             return;
 
-        if (this._watchDog.nameAcquired && this._watchDog.nameOnBus)
+        if (this._watchDog && this._watchDog.nameAcquired && this._watchDog.nameOnBus)
```

Comment:
```suggestion
        if (this?._watchDog.nameAcquired && this._watchDog.nameOnBus)
```

Reply:
Shouldn't it be `this._watchDog?.nameAcquired`? And also do the same check on nameOnBus like
```js
if (this._watchDog?.nameAcquired && this._watchDog?.nameOnBus)
```

Reply:
Sorry, yeah I meant to write `this._watchDog?.nameAcquired`... 

> And also do the same check on nameOnBus

That's not needed because if the 1st one is true, then when checking for the 2nd value we're already sure that's defined.

Reply:
Alright, yeah I got it, it was just for consistency :) will force push tomorrow!